### PR TITLE
zh-cn: update "how_browsers_work"

### PR DESCRIPTION
**Pull request by Ende93**
### Description

zh-cn: update "how_browsers_work"

### CODE REVIEW

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 1]
```diff
@@ -85,11 +85,11 @@ TCP 的“三次握手”技术经常被称为“SYN-SYN-ACK”——更确切
...
-TCP 数据包在传输过程中被分成若干段。由于 TCP 保证数据包的顺序，因此服务器在发送一定数量的数据包后，必须以 ACK 数据包的形式收到客户端的确认。
+在传输过程中，TCP 包被分割成段。由于 TCP 保证了数据包的顺序，因此服务器在发送一定数量的段后，必须从客户端接收一个 ACK 包的确认。
 
-如果服务器在每个网段后都等待 ACK，则会导致客户端频繁发出 ACK，即使在网络负荷较低的情况下也会增加传输时间。
+如果服务器在发送每个段之后都等待 ACK，那么客户端将频繁地发送 ACK，并且可能会增加传输时间，即使在网络负载较低的情况下也是如此。
 
-另一方面，一次性发送过多网段可能会导致这样的问题：在繁忙的网络中，客户端无法接收到网段，只能长时间不停地回应 ACK，服务器不得不不断重新发送网段。
+另一方面，一次发送过多的段会导致在繁忙的网络中客户端无法接收段并且长时间地只会持续发送 ACK，服务器必须不断重新发送段的问题。
 
 为了平衡传输段的数量，{{glossary('TCP slow start', 'TCP 慢启动')}}算法用于逐渐增加传输数据量，直到确定最大网络带宽，并在网络负载较高时减少传输数据量。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:94]: see above

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 2]
```diff
@@ -99,11 +99,11 @@ TCP 数据包在传输过程中被分成若干段。由于 TCP 保证数据包
 
 ## 解析
 
-一旦浏览器收到数据的第一块，它就可以开始解析收到的信息。{{glossary('parse', '“解析”')}}是浏览器将通过网络接收的数据转换为 {{glossary('DOM')}} 和 {{glossary('CSSOM')}} 的步骤，通过渲染器把 DOM 和 CSSOM 在屏幕上绘制成页面。
+一旦浏览器收到第一块数据，它就可以开始解析收到的信息。{{glossary('parse', '“解析”')}}是浏览器将通过网络接收的数据转换为 {{glossary('DOM')}} 和 {{glossary('CSSOM')}} 的步骤，通过渲染器把 DOM 和 CSSOM 在屏幕上绘制成页面。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:102]:
```suggestion
一旦浏览器收到第一个数据分块，它就可以开始解析收到的信息。{{glossary('parse', '“解析”')}}是浏览器将通过网络接收的数据转换为 {{glossary('DOM')}} 和 {{glossary('CSSOM')}} 的步骤，通过渲染器在屏幕上将它们绘制成页面。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 3]
```diff
@@ -119,7 +119,7 @@ DOM 树描述了文档的内容。[`<html>`](/zh-CN/docs/Web/HTML/Element/html)
 
 ### 预加载扫描器
 
-浏览器构建 DOM 树时，这个过程占用了主线程。当这种情况发生时，预加载扫描仪将解析可用的内容并请求高优先级资源，如 CSS、JavaScript 和 web 字体。多亏了预加载扫描器，我们不必等到解析器找到对外部资源的引用来请求它。它将在后台检索资源，以便在主 HTML 解析器到达请求的资源时，它们可能已经在运行，或者已经被下载。预加载扫描仪提供的优化减少了阻塞。
+浏览器构建 DOM 树时，这个过程占用了主线程。同时，预加载扫描器会解析可用的内容并请求高优先级的资源，如 CSS、JavaScript 和网络字体。多亏了预加载扫描器，我们不必等到解析器找到对外部资源的引用时才去请求。它将在后台检索资源，而当主 HTML 解析器解析到要请求的资源时，它们可能已经下载中了，或者已经被下载。预加载扫描仪提供的优化减少了阻塞。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:122]: We will not translate the word "web"

```suggestion
浏览器构建 DOM 树时，这个过程占用了主线程。同时，*预加载扫描器*会解析可用的内容并请求高优先级的资源，如 CSS、JavaScript 和 web 字体。多亏了预加载扫描器，我们不必等到解析器找到对外部资源的引用时才去请求。它将在后台检索资源，而当主 HTML 解析器解析到要请求的资源时，它们可能已经下载中了，或者已经被下载。预加载扫描器提供的优化减少了阻塞。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 4]
```diff
@@ -168,37 +168,37 @@ CSSOM 树包括来自用户代理样式表的样式。浏览器从适用于节
 
 ### 布局
 
-第四步是在渲染树上运行布局以计算每个节点的几何体。*布局*是确定呈现树中所有节点的宽度、高度和位置，以及确定页面上每个对象的大小和位置的过程。回流是对页面的任何部分或整个文档的任何后续大小和位置的确定。
+第四步是在渲染树上运行布局以计算每个节点的几何体。*布局*是确定呈现树中所有节点的尺寸和位置，以及确定页面上每个对象的大小和位置的过程。*回流*是后续过程中对页面的任意部分或整个文档的大小和位置的重新计算。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:171]: https://developer.mozilla.org/zh-CN/docs/Glossary/Reflow

```suggestion
第四步是在渲染树上运行布局以计算每个节点的几何体。*布局*是确定呈现树中所有节点的尺寸和位置，以及确定页面上每个对象的大小和位置的过程。*重排*是后续过程中对页面的任意部分或整个文档的大小和位置的重新计算。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 5]
```diff
@@ -168,37 +168,37 @@ CSSOM 树包括来自用户代理样式表的样式。浏览器从适用于节
...
 
-构建渲染树后，开始布局。渲染树标识显示哪些节点（即使不可见）及其计算样式，但不标识每个节点的尺寸或位置。为了确定每个对象的确切大小和位置，浏览器从渲染树的根开始遍历它。
+渲染树构建完毕后，浏览器就开始布局。渲染树标识了哪些节点会显示（即使不可见）及其计算样式，但不标识每个节点的尺寸或位置。为了确定每个对象的确切大小和位置，浏览器会从渲染树的根开始遍历。
 
-在网页上，大多数东西都是一个盒子。不同的设备和不同的桌面意味着无限数量的不同的视区大小。在此阶段，考虑到视口大小，浏览器将确定屏幕上所有不同框的尺寸。以视口的大小为基础，布局通常从 body 开始，用每个元素的框模型属性排列所有 body 的子孙元素的尺寸，为不知道其尺寸的替换元素（例如图像）提供占位符空间。
+在网页上，大多数东西都是一个盒子。不同的设备和不同的桌面设置意味着无限数量的不同视区大小。在此阶段，根据视口大小，浏览器将确定屏幕上所有盒子的大小。以视口大小为基础，布局通常从 body 开始，设置所有 body 后代的大小，同时给不知道其尺寸的替换元素（例如图像）提供占位符空间，空间大小以相应元素盒模型的属性为准。
 
-第一次确定节点的大小和位置称为布局。随后对节点大小和位置的重新计算称为回流。在我们的示例中，假设初始布局发生在返回图像之前。由于我们没有声明图像的尺寸，因此一旦知道图像的尺寸，就会出现回流。
+第一次确定每个节点的大小和位置称为*布局*。随后对节点大小和位置的重新计算称为*回流*。在我们的示例中，假设初始布局发生在返回图像之前。由于我们没有声明图像的尺寸，因此一旦知道图像的尺寸，就会出现回流。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:177]:
```suggestion
第一次确定每个节点的大小和位置称为*布局*。随后对节点大小和位置的重新计算称为*重排*。在我们的示例中，假设初始布局发生在返回图像之前。由于我们没有声明图像的尺寸，因此一旦知道图像的尺寸，就会出现重排。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 6]
```diff
@@ -168,37 +168,37 @@ CSSOM 树包括来自用户代理样式表的样式。浏览器从适用于节
...
 
 ### 绘制
 
-最后一步是将各个节点绘制到屏幕上，第一次出现的节点称为 [first meaningful paint](/zh-CN/docs/Glossary/First_meaningful_paint)。在绘制或光栅化阶段，浏览器将在布局阶段计算的每个框转换为屏幕上的实际像素。绘画包括将元素的每个可视部分绘制到屏幕上，包括文本、颜色、边框、阴影和替换的元素（如按钮和图像）。浏览器需要非常快地完成这项工作。
+关键渲染路径中的最后一步是将各个节点绘制到屏幕上，其中第一次的绘制被称为[首次有意义的绘制](/zh-CN/docs/Glossary/First_meaningful_paint)。在绘制或光栅化阶段，浏览器将在布局阶段计算的每个盒子转换为屏幕上的实际像素。绘制涉及将元素的每个可见部分绘制到屏幕上，包括文本、颜色、边框、阴影以及按钮和图像等替换元素。浏览器需要以超快的速度执行这个过程。
 
-为了确保平滑滚动和动画，占据主线程的所有内容，包括计算样式，以及回流和绘制，必须让浏览器在 16.67 毫秒内完成。在 2048x1536 分辨率的 iPad 上，有超过 314.5 万像素将被绘制到屏幕上。那是很多像素需要快速绘制。为了确保重绘的速度比初始绘制的速度更快，屏幕上的绘图通常被分解成数层。如果发生这种情况，则需要进行合成。
+为了确保平滑滚动和动画效果，包括计算样式、回流和绘制等占用主线程的所有操作，必须在不超过 16.67 毫秒的时间内完成。在 2048 x 1536 分辨率下，iPad 需要将超过 3,145,000 个像素绘制到屏幕上。这是非常多的像素，必须要非常快速地绘制出来。为了确保重绘能够比初始绘制更快地完成，绘制到屏幕的操作通常被分解成几个图层。如果发生这种情况，浏览器则需要进行合成。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:183]:
```suggestion
为了确保平滑滚动和动画效果，包括计算样式、回流和绘制等占用主线程的所有操作，必须在不超过 16.67 毫秒的时间内完成。在 2048 x 1536 分辨率下，iPad 需要将超过 314.5 万个像素绘制到屏幕上。这是非常多的像素，必须要非常快速地绘制出来。为了确保重绘能够比初始绘制更快地完成，绘制到屏幕的操作通常被分解成几个图层。如果发生这种情况，浏览器则需要进行合成。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 7]
```diff
@@ -168,37 +168,37 @@ CSSOM 树包括来自用户代理样式表的样式。浏览器从适用于节
...
-当页面继续加载资源时，可能会发生回流（回想一下我们迟到的示例图像），回流会触发重新绘制和重新组合。如果我们定义了图像的大小，就不需要重新绘制，只需要重新绘制需要重新绘制的层，并在必要时进行合成。但我们没有包括图像大小！从服务器获取图像后，渲染过程将返回到布局步骤并从那里重新开始。
+当页面继续加载资源时，可能会发生回流（回想一下我们迟到的示例图像），回流会触发重新绘制和重新合成。如果我们定义了图像的大小，就不需要重新绘制，只需要绘制需要重新绘制的层，并在必要时进行合成。但我们并没有定义图像大小！所以从服务器获取图像后，渲染过程将返回到布局步骤并从那里重新开始。
 
 ## 交互
 
-一旦主线程绘制页面完成，你会认为我们已经“准备好了”，但事实并非如此。如果加载包含 JavaScript（并且延迟到 [`onload`](/zh-CN/docs/Web/API/Window/load_event) 事件触发后执行），则主线程可能很忙，无法用于滚动、触摸和其他交互。
+一旦主线程绘制页面完成，你会认为我们已经“准备好了”，但事实并非如此。如果加载包括正确延迟加载的 JavaScript，并且仅在 [`onload`](/zh-CN/docs/Web/API/Window/load_event) 事件触发后执行，那么主线程可能会忙于执行脚本，无法用于滚动、触摸和其他交互操作。
 
-{{glossary('Time to Interactive', '可交互时间（TTI）')}}是测量从第一个请求导致 DNS 查询和 SSL 连接到页面可交互时所用的时间——可交互是 {{glossary('First Contentful Paint')}} 之后的时间点，页面在 50ms 内响应用户的交互。如果主线程正在解析、编译和执行 JavaScript，则它不可用，因此无法及时（小于 50ms）响应用户交互。
+{{glossary('Time to Interactive', '可交互时间（TTI）')}}是测量从第一个请求导致 DNS 查询和 SSL 连接到页面可交互时所用的时间——可交互是在 {{glossary('First Contentful Paint')}} 之后页面在 50ms 内响应用户的交互。如果主线程正在解析、编译和执行 JavaScript，则无法及时（小于 50 毫秒）响应用户交互。
```
Comment (yin1999) [inline files/zh-cn/web/performance/how_browsers_work/index.md:199]: align the unit of time:

```suggestion
{{glossary('Time to Interactive', '可交互时间（TTI）')}}是测量从第一个请求导致 DNS 查询和 SSL 连接到页面可交互时所用的时间——可交互是在{{glossary('First Contentful Paint', '首次内容绘制')}}之后页面在 50ms 内响应用户的交互。如果主线程正在解析、编译和执行 JavaScript，则无法及时（小于 50ms）响应用户交互。
```

File: files/zh-cn/web/performance/how_browsers_work/index.md [hunk 8]
```diff
@@ -7,57 +7,57 @@ l10n:
...
 快速响应的网站提供更好的用户体验。用户期待内容快速加载和交互流畅的 Web 体验。
 
-等待资源加载时间和大部分情况下的浏览器单线程执行是影响 Web 性能的两大主要原因。
+导致 Web 性能问题的原因主要有两种，一是网络延迟，二是大部分情况下的浏览器单线程执行。
 
-等待时间是需要去克服来让浏览器快速加载资源的主要威胁。为了实现快速加载，开发者的目标就是尽可能快的发送请求的信息，至少看起来相当快。网络等待时间是在链路上传送二进制到电脑端所消耗的链路传输时间。Web 性能优化需要做的就是尽可能快的使页面加载完成。
+保障页面快速加载的最大威胁是网络延迟。开发者的目标是尽可能使网站加载速度快，或者至少看起来加载速度超快，以便用户尽快获取所需信息。网络延迟是将字节传输到计算机的时间。Web 性能的意思是我们越是尽快地加载页面，性能就越高。
 
-大部分情况下，浏览器是单线程执行的。为了有流畅的交互，开发者的目标是确保网站从流畅的页面滚动到点击响应的交互性能。渲染时间是关键要素，确保主线程可以完成所有给它的任务并且仍然一直可以处理用户的交互。通过了解浏览器单线程的本质与最小化主线程的责任可以优化 Web 性能，来确保渲染的流畅和交互响应的及时。
+大部分情况下，浏览器被认为是单线程的。也就是说，它们在执行一个任务之前会从头到尾完成一个任务，然后才会接受另一个任务。为了实现流畅的交互，开发者的目标是确保网站的交互性能良好，从流畅的滚动到对触摸的响应。在这里渲染时间至关重要，更短的渲染时间能确保主线程能够完成我们分配给它的所有工作，并且始终可以处理用户的交互。通过理解浏览器的单线程特性，并尽可能和适当地减少主线程的责任，可以提高网页性能，以确保渲染流畅，并且对交互的响应是即时的。
```
[the remark this answers is not in the document]
Comment (Ende93) [reply files/zh-cn/web/performance/how_browsers_work/index.md]: Fixed. New translation is 渲染时间非常关键，我们需要确保主线程能够完成交给它的所有工作，并且始终能够处理用户交互

**Comment by Ende93** [comment]
Thanks for review

**Review by yin1999 — APPROVED** [review]
LGTM. Thank you @Ende93 :heart:

**Pull request merged**